File: backend/global_variables.py
import platform

class GlobalVariables:
    _instance = None
    
    """
    A singleton class to store global variables.
    """

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(GlobalVariables, cls).__new__(cls)
            cls._instance._initialize()
        return cls._instance

    def _initialize(self):
        self._variables = {}

    def set_variable(self, key, value):
        """
        Set a global variable. If the variable already exists, it will be overwritten.
        """
        self._variables[key] = value
    
    def get_variable(self, key, default=None):
        return self._variables.get(key, default)

    def delete_variable(self, key):
        if key in self._variables:
            del self._variables[key]

    def list_variables(self):
        return self._variables.keys()

File: frontend/app/global-variables.ts
// Class to store global variables that can be accessed from any component
export default class GlobalVariables {
  // Singleton instance
  private static instance: GlobalVariables;

  debug_mode: boolean = false;

  private constructor() {}

  // Singleton pattern to ensure only one instance is used
  public static getInstance(): GlobalVariables {
    if (!GlobalVariables.instance) {
      GlobalVariables.instance = new GlobalVariables();
    }

    return GlobalVariables.instance;
  }

  // BELOW FOR CUSTOM GLOBAL VARIABLES

  /* private variables: { [key: string]: any } = {}; */
  /* // Method to set a global variable
  public setVariable(key: string, value: any): void {
    this.variables[key] = value;
  }

  // Method to get a global variable
  public getVariable(key: string): any {
    return this.variables[key];
  }

  // Method to check if a global variable exists
  public hasVariable(key: string): boolean {
    return key in this.variables;
  }

  // Method to remove a global variable
  public removeVariable(key: string): void {
    delete this.variables[key];
  } */
}
